agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp

```diff
@@ -0,0 +1,111 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "ap_autoconfiguration_task.h"
+
+#include "../agent_db.h"
+#include "../backhaul_manager/backhaul_manager_thread.h"
+
+#include <tlvf/ieee_1905_1/tlvAlMacAddress.h>
+#include <tlvf/ieee_1905_1/tlvAutoconfigFreqBand.h>
+#include <tlvf/ieee_1905_1/tlvSearchedRole.h>
+#include <tlvf/wfa_map/tlvSearchedService.h>
+#include <tlvf/wfa_map/tlvSupportedService.h>
+
+#include <beerocks/tlvf/beerocks_message_control.h>
+
+#include <easylogging++.h>
+
+using namespace beerocks;
+using namespace net;
+using namespace son;
+
+static constexpr uint8_t MAX_FAILED_AUTOCONFIG_SEARCH_ATTEMPTS = 20;
+static constexpr uint8_t AUTOCONFIG_DISCOVERY_TIMEOUT_SECONDS  = 1;
+
+#define FSM_MOVE_STATE(radio_iface, new_state)                                                     \
+    ({                                                                                             \
+        LOG(TRACE) << "AP_AUTOCONFIGURATION " << radio_iface                                       \
+                   << " FSM: " << fsm_state_to_string(m_state[radio_iface].state) << " --> "       \
+                   << fsm_state_to_string(new_state);                                              \
+        m_state[radio_iface].state = new_state;                                                    \
+    })
```

Comment:
This kind of macro is used in multiple places. What about moving it to a common header file?

Reply:
It is similar but not the same. This one has unique implementation that fit only to this task.
Anyway, you can suggest opening a task on creating a common FSM infrastructure. I just don't want to do it as part of this PR.